picard/ui/scripteditor.py

```diff
@@ -738,12 +739,15 @@ def _insert_item(self, script_item):
         self.update_scripts_list()
         self.select_script(skip_check=True)
         self.restore_selected_script_index = idx
+        self.save_to_config()
 
     def new_script(self):
         """Add a new (empty) script to the script selection combo box and script list.
         """
         if self.unsaved_changes_confirmation():
-            script_item = FileNamingScript(script=DEFAULT_FILE_NAMING_FORMAT).to_dict()
+            count = len(self.naming_scripts) + 1
+            title = "{0} ({1})".format(DEFAULT_SCRIPT_NAME, count)
```

Comment:
Script title should be translated and I think count based on number of scripts doesn't work well.
I tried something like:

```python
    def new_script_name(self):
        import re
        translated_default_script_name = _(DEFAULT_SCRIPT_NAME)
        escaped = re.escape(translated_default_script_name)
        regex = re.compile('^' + escaped + '(?: \\((\\d+)\\))?$')
        count = 0
        for script in self.naming_scripts.values():
            m = regex.match(script['title'])
            if m:
                num = m.group(1)
                if num is not None:
                    count = max(count, int(num))
                else:
                    count += 1
        return "{0} ({1})".format(translated_default_script_name, count + 1)

    def new_script(self):
        """Add a new (empty) script to the script selection combo box and script list.
        """
        if self.unsaved_changes_confirmation():
            title = self.new_script_name()
            script_item = FileNamingScript(title=title, script=DEFAULT_FILE_NAMING_FORMAT).to_dict()
            self._insert_item(script_item)
```

Much more complex, but it handles translated default script name, and number scripts in a fancier manner.

For example, let's say we start with no script:

- add new script -> Mon script (1)
- rename it -> Mon script
- add new script -> Mon script (2)
- add new script -> Mon script (3)
- delete Mon script (2)
- add new script -> Mon script (4)

I didn't test it fully (we should write tests for it anyway)

Reply:
I like this much better than the simple version I had incorporated, which is what's used for script name generation for tagging scripts.  When I make the change for the naming script, should I also use the same code for generating the new tagging script titles?

Reply:
I'll prepare a PR for you to merge in this one.

Reply:
@rdswift PR ready -> https://github.com/rdswift/picard/pull/10